system/Validation/Validation.php

```diff
@@ -206,8 +206,13 @@ public function check($value, string $rule, array $errors = []): bool
 	 *
 	 * @return boolean
 	 */
-	protected function processRules(string $field, string $label = null, $value, $rules = null, array $data): bool
+	protected function processRules(string $field, string $label = null, $value, $rules = null, array $data = null): bool
 	{
+		if (is_null($data))
```

Comment:
```suggestion
	protected function processRules(string $field, string $label = null, $value, $rules = null, array $data = []): bool
	{
		if (empty($data))
```

Reply:
It has to accept an empty array which is why I added the `null` type. I would definitely have preferred `empty()`.